lib/jxl/modular/transform/palette.h

```diff
@@ -274,9 +274,9 @@ static Status InvPalette(Image &input, uint32_t begin_c, uint32_t nb_colors,
   if (c0 >= input.nb_meta_channels) {
     input.nb_meta_channels--;
   } else {
-    JXL_ASSERT(c0 + nb - 1 < input.nb_meta_channels);
     JXL_ASSERT(static_cast<int>(input.nb_meta_channels) >= 2 - nb);
```

Comment:
I'd appreciate if you can add a comment here of what's going on in this case. Like, why the `if` at all and what's the intention of the assertion. A reviewer should be able to understand the code and reason about whether it is correct.

Reply:
Added comments here and in the similar MetaPalette code to explain the two cases.